src/main/java/com/rocketden/main/model/User.java

```diff
@@ -18,13 +18,21 @@
 @Setter
 public class User {
 
+    // ID used for the database.
     @Id
     @GeneratedValue(strategy = GenerationType.AUTO)
     private Integer id;
 
+    // User ID for user identification and accessible from frontend.
+    @EqualsAndHashCode.Include
+    private String userId;
```

Comment:
Thoughts on changing this to `private String userId = UUID.randomUUID().toString()`? For the following reasons:

1. Guaranteed uniqueness (at least for all intents and purposes)
2. Automatically initialized for every user, so we don't need a manual check for null-ness in our service classes
3. User ID is never shown on the UI so it shouldn't be too big a deal if it's long and ugly

Reply:
This is a good idea! Would I want to ensure that this doesn't override any existing ID, so I could place this in a constructor, and if `userId` is `null`, then use that line above? I'm not sure if this is needed, but wanted to check.

Reply:
Based on the conversation in today's meeting, we'll leave this as a task for another PR, especially because testing this static method could potentially be very difficult. See the [Trello card](https://trello.com/c/EeQrpcem/159-uuid).